From 696c932fa46fe6442f55678a68cac7b2ae38e7c8 Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Thu, 9 Mar 2023 11:15:22 +0300
Subject: [PATCH] fix(base): do not quote $CLINE and $initargs for switch_root

We want word splitting to occur so that the arguments are parsed separately.

Fixes: 2fabaaa6 ("fix(base): shellcheck for modules.d/99base")
---
 modules.d/99base/init.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/99base/init.sh b/modules.d/99base/init.sh
index bd939160..063d8589 100755
--- a/modules.d/99base/init.sh
+++ b/modules.d/99base/init.sh
@@ -328,7 +328,7 @@ if getarg init= > /dev/null; then
     ignoreargs="console BOOT_IMAGE"
     # only pass arguments after init= to the init
     CLINE=${CLINE#*init=}
-    set -- "$CLINE"
+    set -- $CLINE
     shift # clear out the rest of the "init=" arg
     for x in "$@"; do
         for s in $ignoreargs; do
@@ -339,7 +339,7 @@ if getarg init= > /dev/null; then
     unset CLINE
 else
     debug_off # Turn off debugging for this section
-    set -- "$CLINE"
+    set -- $CLINE
     for x in "$@"; do
         case "$x" in
             [0-9] | s | S | single | emergency | auto)
-- 
2.25.1

